File: algorithms/js/singleNumber/singleNumber.js
/*
 * @Source: https://leetcode-cn.com/problems/single-number/
 * @Author: Tian Zhuo
 * @Date: 2019-01-08 21:47:14
 */
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
    let singleNum = 0;
    for (num of nums) {
        singleNum ^= num;
    }
    return singleNum;
};

File: src/main/java/singleNumber/SingleNumber.java
/*
 * @Source: https://leetcode-cn.com/problems/single-number/
 * @Author: Tian Zhuo
 * @Date: 2019-01-08 21:48:03
 */
package singleNumber;

import java.util.Arrays;

public class SingleNumber {
    public int singleNumber1(int[] nums) {
        Arrays.sort(nums);
        for(int i = 0; i < nums.length - 1; i += 2){
            if(nums[i] != nums[i + 1]){
                return nums[i];
            }
        }
        return nums[nums.length - 1];
    }

    public int singleNumber(int[] nums) {
        int singleNum = 0;
        for (int num : nums) {
            singleNum ^= num;
        }
        return singleNum;
    }
}